From 50f02c745e7f6e86ce11c0d5ac8d701aa9d1fca7 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Fri, 27 Feb 2026 03:31:49 +0200
Subject: [PATCH] Support OE-specific "arm" value for CMAKE_SYSTEM_PROCESSOR

OE sets CMAKE_SYSTEM_PROCESSOR to the bare "arm" value instead of
encoding armv7 or something similar. Provide OE-specific fixup for the
cpuinfo code.

Upstream-Status: Inappropriate [oe specific: version-less CMAKE_SYSTEM_PROCESSOR]
Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index aedc98311251..ec078c687710 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,7 +95,7 @@ IF(NOT CMAKE_SYSTEM_PROCESSOR)
       "cpuinfo will compile, but cpuinfo_initialize() will always fail.")
     SET(CPUINFO_SUPPORTED_PLATFORM FALSE)
   ENDIF()
-ELSEIF(NOT CPUINFO_TARGET_PROCESSOR MATCHES "^(i[3-6]86|AMD64|x86(_64)?|armv[5-8].*|aarch64|arm64.*|ARM64.*|riscv(32|64))$")
+ELSEIF(NOT CPUINFO_TARGET_PROCESSOR MATCHES "^(i[3-6]86|AMD64|x86(_64)?|armv[5-8].*|arm|aarch64|arm64.*|ARM64.*|riscv(32|64))$")
   MESSAGE(WARNING
     "Target processor architecture \"${CPUINFO_TARGET_PROCESSOR}\" is not supported in cpuinfo. "
     "cpuinfo will compile, but cpuinfo_initialize() will always fail.")
@@ -189,7 +189,7 @@ IF(CPUINFO_SUPPORTED_PLATFORM)
       src/arm/windows/init-by-logical-sys-info.c
       src/arm/windows/init.c
       src/arm/uarch.c)
-  ELSEIF(CPUINFO_TARGET_PROCESSOR MATCHES "^(armv[5-8].*|aarch64|arm64.*)$" OR IOS_ARCH MATCHES "^(armv7.*|arm64.*)$")
+  ELSEIF(CPUINFO_TARGET_PROCESSOR MATCHES "^(armv[5-8].*|aarch64|arm64.*)$" OR CPUINFO_TARGET_PROCESSOR STREQUAL "arm" OR IOS_ARCH MATCHES "^(armv7.*|arm64.*)$")
     LIST(APPEND CPUINFO_SRCS
       src/arm/uarch.c
       src/arm/cache.c)
@@ -201,7 +201,7 @@ IF(CPUINFO_SUPPORTED_PLATFORM)
         src/arm/linux/chipset.c
         src/arm/linux/midr.c
         src/arm/linux/hwcap.c)
-      IF(CMAKE_SYSTEM_PROCESSOR MATCHES "^armv[5-8]")
+      IF(CMAKE_SYSTEM_PROCESSOR MATCHES "^armv[5-8]" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "arm" )
         LIST(APPEND CPUINFO_SRCS src/arm/linux/aarch32-isa.c)
         IF(CMAKE_SYSTEM_NAME STREQUAL "Android" AND ANDROID_ABI STREQUAL "armeabi")
           SET_SOURCE_FILES_PROPERTIES(src/arm/linux/aarch32-isa.c PROPERTIES COMPILE_FLAGS -marm)
-- 
2.47.3

